az-digital/az_quickstart
# Closes #574 Adding az_menu block_content type.

## Description


## Related issues
Closes #574
Please review https://git.drupalcode.org/project/field_menu/-/merge_requests/2/diffs

## How to test

Enable the `az_block_types_menu` module. (Not enabled by default)
<img width="1642" alt="image" src="https://user-images.githubusercontent.com/1023167/167746232-b0d1ae26-ac96-4869-947b-e8c24f4b07a4.png">
Go to: /admin/content/blocks
<img width="738" alt="image" src="https://user-images.githubusercontent.com/1023167/167746139-37cf4f99-f753-457e-b2f5-a8a9d68404b3.png">
See the menu option.
Go to `/admin/structure/block/block-content/manage/az_menu`
Review settings 
Go to `/admin/structure/block/block-content/manage/az_menu/fields`
Review settings 
Go to `/admin/structure/block/block-content/manage/az_menu/display`
Review settings 
Go to `/admin/structure/block/block-content/manage/az_menu/auto-label`
Review settings 
Either add a new menu at `/admin/structure/menu/add`
or 
Go to `/block/add/az_menu`
 
Select your new menu or any of the other existing menus in the Root field of the menu field set.
You should be able to select any level of any menu.
Save your new block

Go to `/node/add/az_flexible_page`
Add a title and a new "text" paragraph item.
Then click the block embed icon in the editor
Search for your block by the description you added when creating a new block.
When. found, click next and choose some options.
Your menu block should now be embedded within a text area.

Now that you know how to add a new menu block that can be embedded, be creative and add menu blocks to any field that allows embedding blocks.

Additionally you should be able to add the new block via the traditional block layout page.

Uninstall `az_block_types_menu` module to delete all of your tests.
Reinstall `az_block_types_menu` module to ensure it is possible.

## Types of changes
### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
- Minor release changes

### Drupal core
- Patch release changes
- Minor release changes

### Drupal contrib projects
- Patch release changes
- Minor release changes

## Checklist
